FileSystemManager/Controllers/FileSystemController.cs

```diff
@@ -287,162 +289,249 @@ public IActionResult OpenFile(string path)
         }
 
         [HttpDelete]
-        public IActionResult DeleteItem(string path)
+        public async Task<IActionResult> DeleteItem(string path)
         {
             if (string.IsNullOrEmpty(path))
             {
                 return BadRequest("Invalid path");
             }
-
-            var decodedPath = System.Net.WebUtility.UrlDecode(path);
-            var fullPath = Path.Combine(rootPath, decodedPath.TrimStart('/'));
-
-            try
+            using (var transaction = await _context.Database.BeginTransactionAsync())
             {
-                if (Directory.Exists(fullPath))
+                try
                 {
-                    Directory.Delete(fullPath, true);
-                }
-                else if (System.IO.File.Exists(fullPath))
-                {
-                    System.IO.File.Delete(fullPath);
+                    var decodedPath = WebUtility.UrlDecode(path).Replace('\\', '/');
+                    var fullPath = Path.Combine(rootPath, decodedPath).Replace('\\', '/');
+                    var recycleBinPath = Path.Combine(_recycleBinPath, Path.GetFileName(fullPath)).Replace('/', '\\');
+                    Directory.CreateDirectory(Path.GetDirectoryName(recycleBinPath));
+                    if (Directory.Exists(fullPath))
+                    {
+                        Directory.Move(fullPath, recycleBinPath);
+                    }
+                    else if (System.IO.File.Exists(fullPath))
+                    {
+                        if (System.IO.File.Exists(recycleBinPath))
+                        {
+                            System.IO.File.Delete(recycleBinPath);
+                        }
+                        System.IO.File.Move(fullPath, recycleBinPath);
+                    }
+                    else
+                    {
+                        return NotFound();
+                    }
+                    // Normalize the path for consistent database querying
+                    var normalizedPath = decodedPath;
+                    var allFileMetadata = _context.FileMetadata.AsEnumerable();
+                    var fileMetadata = allFileMetadata
+                        .FirstOrDefault(fm => fm.FilePath.Replace('\\', '/') == normalizedPath);
+                    if (fileMetadata != null)
+                    {
+                        fileMetadata.IsDeleted = true;
+                        fileMetadata.DeletedPath = recycleBinPath;
+                        fileMetadata.DateModified = DateTime.UtcNow;
+                        _context.Update(fileMetadata);
+                        await _context.SaveChangesAsync();
+                    }
+                    await transaction.CommitAsync();
+                    return Ok();
                 }
-                else
+                catch (Exception ex)
                 {
-                    return NotFound();
+                    await transaction.RollbackAsync();
```

Comment:
The transaction fails but directory locally was changed, should it not reverse deletion for local changes?

Reply:
Added